webui/react/src/e2e/models/components/DeterminedAuth.ts

```diff
@@ -0,0 +1,24 @@
+import { BaseComponent } from 'e2e/models/BaseComponent';
+
+/**
+ * Returns a representation of the DeterminedAuth component.
+ *
+ * @remarks
+ * This constructor represents the contents in src/components/DeterminedAuth.tsx.
+ *
+ * @param {Object} obj
+ * @param {implementsGetLocator} obj.parent - The parent used to locate this DeterminedAuth
+ * @param {string} [obj.selector] - Used instead of `defaultSelector`
+ */
+export class DeterminedAuth extends BaseComponent {
+  static defaultSelector: string = 'Form[data-test=authForm]';
+  override readonly defaultSelector: string = DeterminedAuth.defaultSelector;
+
+  readonly form: BaseComponent = new BaseComponent({ parent: this, selector: 'form' });
```

Comment:
I really like this. it feels very declarative!

Reply:
teamwork!